src/java/com/puppetlabs/http/client/impl/PersistentAsyncHttpClient.java

```diff
@@ -15,13 +15,16 @@
     private CloseableHttpAsyncClient client;
     private MetricRegistry metricRegistry;
     private String metricNamespace;
+    private Boolean useURLMetrics;
 
     public PersistentAsyncHttpClient(CloseableHttpAsyncClient client,
                                      MetricRegistry metricRegistry,
-                                     String metricNamespace) {
+                                     String metricNamespace,
+                                     Boolean useURLMetrics) {
```

Comment:
I'd prefer the `boolean` primitive to avoid boxing/unboxing and null checks that come with the `Boolean` class.  Seems like wherever we use this we're expecting it to just have a `true` or `false` value, right?